### PR TITLE
[Storybook8] Migrate React hooks and best practices to V8

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
[Storybook8] Migrate React hooks and best practices to V8

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3785956
https://skype.visualstudio.com/SPOOL/_workitems/edit/3785960

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook8/stories/StatefulClient/ReactHooks/UsePropsForHook.mdx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="Stateful Client/React Hooks/UsePropsFor" />
```
Comment (JoshuaLai) [inline packages/storybook8/stories/StatefulClient/ReactHooks/UsePropsForHook.mdx:3]:
```suggestion
<Meta title="Stateful Client/React Hooks/usePropsFor" />
```
To be more accurate with the API should we use camel case in this case?

File: packages/storybook8/stories/StatefulClient/ReactHooks/UseSelector.mdx [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="Stateful Client/React Hooks/UseSelector" />
```
Comment (JoshuaLai) [inline packages/storybook8/stories/StatefulClient/ReactHooks/UseSelector.mdx:3]:
```suggestion
<Meta title="Stateful Client/React Hooks/useSelector" />
```

Same comment here

**Review by JoshuaLai — APPROVED** [review]
Just two nitpick comments on the naming of the side panel

**Review by carocao-msft — APPROVED** [review]

File: packages/storybook8/stories/StatefulClient/ReactHooks/UseSelector.mdx (hunk 2 above)
Comment (prabhjot-msft) [reply packages/storybook8/stories/StatefulClient/ReactHooks/UseSelector.mdx:3]: In current storybook we have these hooks starting with capital, keeping it same as of now, will update later if feel we should switch 
Screenshot 2024-07-26 at 10 10 25 AM

**Pull request merged**